CMake/SiloFindPython.cmake

```diff
@@ -0,0 +1,72 @@
+# are met:
+# 
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the disclaimer below.
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the disclaimer (as noted
+#      below) in the documentation and/or other materials provided with
+#      the distribution.
+#    * Neither the name of the LLNS/LLNL nor the names of its
+#      contributors may be used to endorse or promote products derived
+#      from this software without specific prior written permission.
+# 
+# THIS SOFTWARE  IS PROVIDED BY  THE COPYRIGHT HOLDERS  AND CONTRIBUTORS
+# "AS  IS" AND  ANY EXPRESS  OR IMPLIED  WARRANTIES, INCLUDING,  BUT NOT
+# LIMITED TO, THE IMPLIED  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A  PARTICULAR  PURPOSE ARE  DISCLAIMED.  IN  NO  EVENT SHALL  LAWRENCE
+# LIVERMORE  NATIONAL SECURITY, LLC,  THE U.S.  DEPARTMENT OF  ENERGY OR
+# CONTRIBUTORS BE LIABLE FOR  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR  CONSEQUENTIAL DAMAGES  (INCLUDING, BUT NOT  LIMITED TO,
+# PROCUREMENT OF  SUBSTITUTE GOODS  OR SERVICES; LOSS  OF USE,  DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER  IN CONTRACT, STRICT LIABILITY,  OR TORT (INCLUDING
+# NEGLIGENCE OR  OTHERWISE) ARISING IN  ANY WAY OUT  OF THE USE  OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# 
+# This work was produced at Lawrence Livermore National Laboratory under
+# Contract No.  DE-AC52-07NA27344 with the DOE.
+# 
+# Neither the  United States Government nor  Lawrence Livermore National
+# Security, LLC nor any of  their employees, makes any warranty, express
+# or  implied,  or  assumes  any  liability or  responsibility  for  the
+# accuracy, completeness,  or usefulness of  any information, apparatus,
+# product, or  process disclosed, or  represents that its use  would not
+# infringe privately-owned rights.
+# 
+# Any reference herein to  any specific commercial products, process, or
+# services by trade name,  trademark, manufacturer or otherwise does not
+# necessarily  constitute or imply  its endorsement,  recommendation, or
+# favoring  by  the  United  States  Government  or  Lawrence  Livermore
+# National Security,  LLC. The views  and opinions of  authors expressed
+# herein do not necessarily state  or reflect those of the United States
+# Government or Lawrence Livermore National Security, LLC, and shall not
+# be used for advertising or product endorsement purposes.
+#
+##############################################################################
+
+###
+# Uses find_package to find python
+#   If SILO_PYTHON_DIR is defined, uses it to tell CMake where to look.
+###
+
+
+if(DEFINED SILO_PYTHON_DIR)
+    # help CMake find the specified python
+    set(Python_ROOT ${SILO_PYTHON_DIR})
+endif()
+
+find_package(Python COMPONENTS Interpreter Development)
```

Comment:
Will this work for python 2 or 3? Silo's python module is designed to work with both python 2 and python 3 so it shouldn't matter which version of Python we ultimately get. I just wanna make sure this can find whatever is available. On AutoTools, I would just wind up using whatever was in the user's path.

Reply:
this version of find_package for python will find either version, with Python 3 preferred. But if system only has python 2 or the Python_ROOT  points to python2 install, it should return python 2.